### PR TITLE
Add files for pseudobulk workflow using decoupler and edgeR

### PR DESCRIPTION
**Pull request by dianichj**
This workflow utilizes the decoupler tool in Galaxy to generate pseudobulk counts from an AnnData file obtained from scRNA-seq analysis. Following the pseudobulk step, differential expression genes (DEG) are calculated using the edgeR tool. The workflow also includes data sanitation steps to ensure smooth operation of edgeR, minimizing potential issues. Additionally, a Volcano plot tool is used to visualize the results after the DEG analysis.

### CODE REVIEW

**Comment by dianichj** [comment]
@galaxyproject/iwc Hi there! can someone help me by reviewing this one? Thanks lots

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      class: File
+      path: test-data/contrasts_files.txt
+      filetype: txt
+    'Pseudo-bulk: Fields to merge': null
+    Groupy column: cell_type
+    Sample key column: individual
+    Name Your Raw Counts Layer: counts
+    Factor fields: disease
+    Gene symbol column: gene_symbol
+    Formula: '~ 0 + disease '
```
Comment (mvdbeek) [inline workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml:17]:
```suggestion
    Formula: '~ 0 + disease'
```

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  outputs:
+    'pseudobulk_count_matrix':
```
Comment (mvdbeek) [inline workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml:19]: Ca you give the outputs human readable labels, i.e. no underscores ?

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        delta: 2000 
+    'Filtered by expression':
+      element_test:
+        has_size: 23490
+        delta: 2000 
+    'Report Results: HTML File':
+      element_test:
+        has_size: 531761
+        delta: 25000
+    'Tables: DEG ':
```
Comment (mvdbeek) [inline workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml:35]:
```suggestion
    'Tables: DEG':
```
This needs changing in the workflow output label as well.

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/test-data/contrasts_files.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Header
```
Comment (mvdbeek) [inline workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/test-data/contrasts_files.txt:1]: How does a user know what to put in here ? Can we avoid this file ? Alternatively can this be a text input parameter ?

**Comment by pcm32** [comment]
Please cite here (or in the tutorial, whatever is best) the original workflow on which this is based (Persist-seq Pseudo-bulk workflow). I make my changes there, and this seems already like quite an old version of it. Not sure if you prefer that one to be on Github or on a Galaxy instance (to point to).

**Comment by dianichj** [comment]
> Please cite here (or in the tutorial, whatever is best) the original workflow on which this is based (Persist-seq Pseudo-bulk workflow). I make my changes there, and this seems already like quite an old version of it. Not sure if you prefer that one to be on Github or on a Galaxy instance (to point to).

Yes, of course, we can cite the original workflow in the pseudo-bulk tutorial. Please let me know where it is published, and I will make sure to include it once the tutorial is created. As a co-author of this workflow, your contribution is very important. This submitted workflow is not intended to be an older version of the Persist-seq workflow. Instead, it has been adapted to follow the PBMCs clustering with Scanpy tutorial in Galaxy. We also made adjustments to inputs, parameters, etc...to make it user-friendly. Currently, based on Marius' suggestions and feedback, we are working on further automating it.

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/test-data/contrasts_files.txt (hunk 4 above)
Comment (dianichj) [reply workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/test-data/contrasts_files.txt:1]: We are working on it ;)! Thank you!

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml (hunk 2 above)
Comment (dianichj) [reply workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml]: Hi Marius! Thanks a lot for your feedback. I get this error when trying to give a more human readable name, I think it is not possible. Is there a way to do it or should I just keep it as it was? Thanks a lot!  .. ERROR: Test found for unknown workflow output [Pseudobulk Count Matrix], workflow outputs [['_anonymous_output_1', '_anonymous_output_2', '_anonymous_output_3', '_anonymous_output_4', '_anonymous_output_5', '_anonymous_output_6', '_anonymous_output_7', 'pseudobulk_count_matrix', 'Pseudobulk Plot', 'Filtered by expression', 'Tables: DEG ', 'Report Results: HTML File', 'Tables for volcano plot', 'Volcano Plot on input dataset(s): PDF']]
.. ERROR: Test found for unknown workflow output [Tables for Volcano Plot], workflow outputs [['_anonymous_output_1', '_anonymous_output_2', '_anonymous_output_3', '_anonymous_output_4', '_anonymous_output_5', '_anonymous_output_6', '_anonymous_output_7', 'pseudobulk_count_matrix', 'Pseudobulk Plot', 'Filtered by expression', 'Tables: DEG ', 'Report Results: HTML File', 'Tables for volcano plot', 'Volcano Plot on input dataset(s): PDF']]

Comment (mvdbeek) [reply workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml:15]: You have to update the labels in the workflow as well.

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/test-data/contrasts_files.txt (hunk 4 above)
Comment (dianichj) [reply workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/test-data/contrasts_files.txt:1]: Hi @mvdbeek ! We added an AWK code step. No file needed anymore =D

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR.ga [hunk 5]
```diff
@@ -0,0 +1,1022 @@
...
+            "name": "Pavankumar Videm"
+        },
+        {
+            "class": "Person",
+            "identifier": "0000-0002-9856-1679",
+            "name": "Pablo Moreno"
+        }
+    ],
+    "format-version": "0.1",
+    "license": "CC-BY-4.0",
```
Comment (nilchia) [inline workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR.ga:23]:
```suggestion
    "license": "CC-BY-4.0",
    "release": "0.1",
```

**Review by nilchia — CHANGES_REQUESTED** [review]

**Comment by dianichj** [comment]
Hi, could someone help me here? I don't know now what the problem could be. I run it locally again after @nilchia help me to add the missing release in the workflow and it worked well. It generated the json test files without any issues. I would appreciate someone's help. Thank you!!

**Comment by dianichj** [comment]
Thanks so much for everyone who contributed to fixing and improving my workflow pull request! I think if someone could review it again, it might be ready to merged: @galaxyproject/iwc

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml [hunk 6]
```diff
@@ -0,0 +1,50 @@
+- doc: Test outline for pseudo-bulk_edgeR
+  job:
+    Source AnnData file:
+      class: File
+      location: https://zenodo.org/records/13929549/files/Source%20AnnData%20file.h5ad
+      filetype: h5ad
+    'Pseudo-bulk: Fields to merge': null
+    Groupy column: cell_type
```
Comment (mvdbeek) [inline workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR-tests.yml:8]:
```suggestion
    Group by column: cell_type
```

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR.ga [hunk 7]
```diff
@@ -0,0 +1,1023 @@
...
+    "release": "0.1",
+    "name": "pseudo-bulk_edgeR",
```
Comment (mvdbeek) [inline workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR.ga:25]:
```suggestion
    "name": "Differential gene expression for single-cell data using pseudo-bulk counts with edgeR",
```
Or something like this ? That's what the users will see in galaxy and https://iwc.galaxyproject.org/

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR.ga [hunk 8]
```diff
@@ -0,0 +1,1023 @@
...
+        "2": {
+            "annotation": "Typically, the column in obs that you want to use for comparisons later (the main contrast field) should be specified here. This column will also be used for plotting the pseudo-bulk samples, showing the number of counts and cells.",
+            "content_id": null,
+            "errors": null,
+            "id": 2,
+            "input_connections": {},
+            "inputs": [
+                {
+                    "description": "Typically, the column in obs that you want to use for comparisons later (the main contrast field) should be specified here. This column will also be used for plotting the pseudo-bulk samples, showing the number of counts and cells.",
+                    "name": "Groupy column"
```
Comment (mvdbeek) [inline workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR.ga:99]:
```suggestion
                    "name": "Group by column"
```

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR.ga [hunk 9]
```diff
@@ -0,0 +1,1023 @@
...
+                }
+            ],
+            "label": "Groupy column",
```
Comment (mvdbeek) [inline workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR.ga:102]:
```suggestion
            "label": "Group by column",
```

File: workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR.ga [hunk 10]
```diff
@@ -0,0 +1,1023 @@
+{
+    "a_galaxy_workflow": "true",
+    "annotation": "Pseudobulk",
```
Comment (mvdbeek) [inline workflows/scRNAseq/pseudobulk-worflow-decoupler-edger/pseudo-bulk_edgeR.ga:3]:
```suggestion
    "annotation": "This workflow uses the decoupler tool in Galaxy to generate pseudobulk counts from an annotated AnnData file obtained from scRNA-seq analysis. Following the pseudobulk step, differential expression genes (DEG) are calculated using the edgeR tool. The workflow also includes data sanitation steps to ensure smooth operation of edgeR and minimizing potential issues. Additionally, a Volcano plot tool is used to visualize the results after the DEG analysis.",
```

**Review by mvdbeek — APPROVED** [review]
This looks really good, just some small cosmetic fixes here.

**Review by bgruening — APPROVED** [review]

**Pull request merged**